### ISSUE TITLE
Page Builder module has very broad slider css rule that overrides all buttons on the site

### ISSUE THREAD

**Issue by mzenner1** [magento/magento2#34527]
### Summary (*)
vendor/magento/module-page-builder/view/frontend/web/css/source/content-type/slider/_default.less
This file has an un-nested less rule for "a.button" which conflicts with all anchor tags with a class of button on an entire website (not limited to just page builder pieces). 

### Examples (*)
Line is highlighted below
& when (@media-common = true) {
    div[data-content-type='slider'] {
        visibility: hidden;

        &.slick-initialized {
            visibility: visible;
        }

        .slick-list,
        .slick-track,
        .slick-slide {
            min-height: inherit;

            > div {
                line-height: 0;
                min-height: inherit;
                overflow: hidden;
                width: 100%;
            }
        }
    }

    **a.button {
        -moz-appearance: button;
        -webkit-appearance: button;
        appearance: button;
        color: initial;
        padding: 10px;
        text-decoration: none;
    }**
}

### Proposed solution
Proposed solution is to only apply this line if the button is actually within a page builder slider template piece.  See below:

& when (@media-common = true) {
    div[data-content-type='slider'] {
        visibility: hidden;

        &.slick-initialized {
            visibility: visible;
        }

        .slick-list,
        .slick-track,
        .slick-slide {
            min-height: inherit;

            > div {
                line-height: 0;
                min-height: inherit;
                overflow: hidden;
                width: 100%;
            }
        }

        a.button {
            -moz-appearance: button;
            -webkit-appearance: button;
            appearance: button;
            color: initial;
            padding: 10px;
            text-decoration: none;
        }
    }
}

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users with no workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [X] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by engcom-Hotel** [comment]
Hello @mzenner1,

Thank you for your report.

To speed up the processing of this issue, please provide us with the following information:

- Summary of the issue
- Information on your environment
- Steps to reproduce
- Expected and actual results

Thanks

**Comment by mzenner1** [comment]
The initial comment has all the relevant details needed to troubleshoot the issue. It's just a matter of a stray "a.button" in a LESS file that is not nested inside a page builder element. As a result, this line of css affects all buttons on a whole website, not just page builder pieces

**Comment by engcom-Hotel** [comment]
Hi @mzenner1,

I understood your reference after checking the less file which you referred to in the description. So based on it, can you please below additional details:
1. Please confirm if this issue is happening with the Composer-based 2.4.x version or is it happening on 2.4-develop as well?
2. If you can share some screenshots like how this particular less file code is affecting whole website buttons? It’ll be faster to reproduce then.

Accordingly, I’ll proceed with this issue.

Thanks

**Comment by mzenner1** [comment]
1. This issue is happening on 2.4.x.  If the same line of css appears on the 2.4-develop branch then it is happening there too.  

2. There is no easy way to provide a screenshot of this issue.  All this really boils down to is that you have a very broad css rule that will apply css to any and all anchor tags that have a class of "button"

a.button {
        -moz-appearance: button;
        -webkit-appearance: button;
        appearance: button;
        color: initial;
        padding: 10px;
        text-decoration: none;
    }

The rule in question in above. The example of how this will affect a website is that it will now adding a 10 pixel padding to all a.button instances on the whole website. The purpose of that file though is to give page builder slider pieces some basic styling.  This rule does not just apply to page builder sliders though.  It applies to everything.  My original post on this thread has the proposed solution to apply this to just page builder slider pieces

**Comment by engcom-Echo** [comment]
Hi @mzenner1,

Thank you for your input. But I am not able to find any affect of Page Builder module on the website storefront or in any of the button as you suggested. Neither I am able to find 10px padding in any button after adding Page Builder button changes. 

I need to reproduce this issue if I want to confirm it. So can you please share more details like video of the issue happening if not screenshots which can help me in reproducing this ?

Thanks

**Comment by mzenner1** [comment]
Attached is a screenshot of the issue.  Using a standard anchor tag with a class of "button" has some of its css rules wiped out/overridden by the broad page builder css rule of "a.button"


![image](https://user-images.githubusercontent.com/15268024/148068812-9ab3393a-52c3-4e74-b882-1ac0f3e04207.png)


This is not part of any page builder piece. It's a custom phtml file loaded on the home page but it is being affected by the page builder css.

**Comment by engcom-Echo** [comment]
Hi @mzenner1,

Thank you for providing this info.

Verified the issue and issue is reproducible on 2.4-develop with Page Builder Module. Below are the steps I did to reproduce this:
1. Enabled Page Builder from Admin Panel.
2. Created simple custom module.
3. Added anchor tag with class button.
4. The css from Page Builder Slider(app/code/Magento/PageBuilder/view/frontend/web/css/source/content-type/slider/_default.less) is affecting this custom created anchor tag. Attached screenshot of it.

<img width="1394" alt="Screenshot 2022-01-07 at 9 49 35 AM" src="https://user-images.githubusercontent.com/51681435/148491814-e8782bab-96d2-44fd-a169-40b4de44b923.png">

This css was meant only for Page Builder Slider but it is affecting all other css element with anchor tag and class button, so confirming this issue.

Thanks
